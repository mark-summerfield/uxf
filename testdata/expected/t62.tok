Import Str("ttype-test.uxi")
TClassBegin ttype=<point>
  Field Str("x") vtype=int
  Field Str("y") vtype=int
  Field Str("z") vtype=int
TClassEnd
TClassBegin ttype=<Files>
  Field Str("Kind")
  Field Str("Filename")
TClassEnd
MapBegin
  Str("files")
  TableBegin ttype=<Files>
    Str("current")
    Str("/home/mark/app/rs/tlm/PlaylistsTest.tlm")
    Str("recent1")
    Str("/home/mark/app/rs/tlm/PlaylistsTest.tlm")
    Str("recent2")
    Str("/home/mark/data/playlists-all.tlm")
  TableEnd
  Str("More Tests")
  ListBegin # A list of name value pairs; normally would be a map
    Str("one")
    TableBegin ttype=<IPv4>
      Int(141)
      Int(179)
      Int(19)
      Int(9)
    TableEnd
    Str("many")
    TableBegin ttype=<IPv4>
      Int(172)
      Int(16)
      Int(254)
      Int(1)
      Int(192)
      Int(0)
      Int(2)
      Int(235)
      Int(127)
      Int(0)
      Int(0)
      Int(1)
    TableEnd
    Str("red")
    TableBegin ttype=<rgb>
      Int(255)
      Int(0)
      Int(0)
    TableEnd
    Str("semi-transparent blue")
    TableBegin ttype=<rgba>
      Int(0)
      Int(0)
      Int(255)
      Int(80)
    TableEnd
    Str("points")
    TableBegin ttype=<point>
      Int(-1)
      Int(2)
      Int(3)
      Int(-4)
      Int(5)
      Int(6)
      Int(-7)
      Int(8)
      Int(9)
    TableEnd
  ListEnd
MapEnd
Eof
----------------------------------------
