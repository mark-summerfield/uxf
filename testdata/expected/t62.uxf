uxf 1.0 TLM Config
!ttype-test.uxi
=Files Kind Filename
=point x:int y:int z:int
{
  <More Tests>
    [#<A list of name value pairs; normally would be a map> 
      <one> (IPv4 141 179 19 9) <many>
      (IPv4
        172 16 254 1
        192 0 2 235
        127 0 0 1
      ) <red> (rgb 255 0 0) <semi-transparent blue> (rgba 0 0 255 80) <points>
      (point
        -1 2 3
        -4 5 6
        -7 8 9
      )
    ]
  <files>
    (Files
      <current> </home/mark/app/rs/tlm/PlaylistsTest.tlm>
      <recent1> </home/mark/app/rs/tlm/PlaylistsTest.tlm>
      <recent2> </home/mark/data/playlists-all.tlm>
    )
}
