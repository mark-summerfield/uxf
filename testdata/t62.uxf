uxf 1 TLM Config
! ttype-test.uxi
= point x:int y:int z:int
= Files Kind Filename
{
    <files> (Files
        <current> </home/mark/app/rs/tlm/PlaylistsTest.tlm>
        <recent1> </home/mark/app/rs/tlm/PlaylistsTest.tlm>
        <recent2> </home/mark/data/playlists-all.tlm>
    )
    <More Tests> [#<A list of name value pairs; normally would be a map>
        <one> (IPv4 141 179 19 9)
        <many> (IPv4
                172 16 254 1
                192 0 2 235
                127 0 0 1)
        <red> (rgb 255 0 0)
        <semi-transparent blue> (rgba 0 0 255 80)
        <points> (point -1 2 3 -4 5 6 -7 8 9)
    ]
}
