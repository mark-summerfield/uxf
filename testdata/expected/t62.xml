<?xml version="1.0" ?>
<uxf version="1.0" custom="TLM Config">
  <imports>
    <import filename="ttype-test.uxi"/>
  </imports>
  <ttypes>
    <ttype name="Files">
      <field name="Kind"/>
      <field name="Filename"/>
    </ttype>
    <ttype name="IPv4">
      <field name="A" vtype="int"/>
      <field name="B" vtype="int"/>
      <field name="C" vtype="int"/>
      <field name="D" vtype="int"/>
    </ttype>
    <ttype name="pair" comment="a pair of any two values">
      <field name="first"/>
      <field name="second"/>
    </ttype>
    <ttype name="point">
      <field name="x" vtype="int"/>
      <field name="y" vtype="int"/>
      <field name="z" vtype="int"/>
    </ttype>
    <ttype name="rgb" comment="0-255 per component">
      <field name="red" vtype="int"/>
      <field name="green" vtype="int"/>
      <field name="blue" vtype="int"/>
    </ttype>
    <ttype name="rgba" comment="0-255 per component">
      <field name="red" vtype="int"/>
      <field name="green" vtype="int"/>
      <field name="blue" vtype="int"/>
      <field name="alpha" vtype="int"/>
    </ttype>
  </ttypes>
  <map>
    <key>
      <str>files</str>
    </key>
    <value>
      <table name="Files">
        <row>
          <str>current</str>
          <str>/home/mark/app/rs/tlm/PlaylistsTest.tlm</str>
        </row>
        <row>
          <str>recent1</str>
          <str>/home/mark/app/rs/tlm/PlaylistsTest.tlm</str>
        </row>
        <row>
          <str>recent2</str>
          <str>/home/mark/data/playlists-all.tlm</str>
        </row>
      </table>
    </value>
    <key>
      <str>More Tests</str>
    </key>
    <value>
      <list comment="A list of name value pairs; normally would be a map">
        <str>one</str>
        <table name="IPv4">
          <row>
            <int v="141"/>
            <int v="179"/>
            <int v="19"/>
            <int v="9"/>
          </row>
        </table>
        <str>many</str>
        <table name="IPv4">
          <row>
            <int v="172"/>
            <int v="16"/>
            <int v="254"/>
            <int v="1"/>
          </row>
          <row>
            <int v="192"/>
            <int v="0"/>
            <int v="2"/>
            <int v="235"/>
          </row>
          <row>
            <int v="127"/>
            <int v="0"/>
            <int v="0"/>
            <int v="1"/>
          </row>
        </table>
        <str>red</str>
        <table name="rgb">
          <row>
            <int v="255"/>
            <int v="0"/>
            <int v="0"/>
          </row>
        </table>
        <str>semi-transparent blue</str>
        <table name="rgba">
          <row>
            <int v="0"/>
            <int v="0"/>
            <int v="255"/>
            <int v="80"/>
          </row>
        </table>
        <str>points</str>
        <table name="point">
          <row>
            <int v="-1"/>
            <int v="2"/>
            <int v="3"/>
          </row>
          <row>
            <int v="-4"/>
            <int v="5"/>
            <int v="6"/>
          </row>
          <row>
            <int v="-7"/>
            <int v="8"/>
            <int v="9"/>
          </row>
        </table>
      </list>
    </value>
  </map>
</uxf>
